src/fastaq_handler.cpp

```diff
@@ -26,21 +25,26 @@ FastaqHandler::FastaqHandler(const std::string& filepath)
     this->inbuf = kseq_init(this->fastaq_file);
 }
 
-FastaqHandler::~FastaqHandler() { close(); }
+FastaqHandler::~FastaqHandler() {
+    if (!this->is_closed()) {
+        close();
+    }
+}
 
-bool FastaqHandler::eof() const { return (this->read_status == -1); }
+bool FastaqHandler::eof() const { return ks_eof(this->inbuf->f); }
 
 void FastaqHandler::get_next()
 {
-    this->read_status = kseq_read(this->inbuf);
+    int read_status = kseq_read(this->inbuf);
 
-    if (this->eof()) {
+    bool no_more_reads_available = read_status == -1;
+    if (no_more_reads_available) {
```

Comment:
`eof()` does not work here anymore because of changes to it, this is basically the previous `eof()` code, which works fine here